pkg/collector/corechecks/systemd/systemd.go

```diff
@@ -0,0 +1,448 @@
+func computeUptime(activeState string, activeEnterTimestampMicroSec uint64, nanoNow int64) int64 {
+	if activeState != unitActiveState {
+		return 0
+	}
+	uptime := nanoNow/1000 - int64(activeEnterTimestampMicroSec)
+	if uptime < 0 {
+		return 0
+	}
+	return uptime
+}
+
+func getPropertyUint64(properties map[string]interface{}, propertyName string) (uint64, error) {
+	prop, ok := properties[propertyName]
+	if !ok {
+		return 0, fmt.Errorf("Property %s not found", propertyName)
+	}
+	switch typedProp := prop.(type) {
+	case uint:
+		return uint64(typedProp), nil
+	case uint32:
+		return uint64(typedProp), nil
+	case uint64:
+		return typedProp, nil
+	}
+	return 0, fmt.Errorf("Property %s (%T) cannot be converted to uint64", propertyName, prop)
+}
+
+func getPropertyString(properties map[string]interface{}, propertyName string) (string, error) {
+	prop, ok := properties[propertyName]
+	if !ok {
+		return "", fmt.Errorf("Property %s not found", propertyName)
+	}
+	propValue, ok := prop.(string)
+	if !ok {
+		return "", fmt.Errorf("Property %s (%T) cannot be converted to string", propertyName, prop)
+	}
+	return propValue, nil
+}
+
+func getPropertyBool(properties map[string]interface{}, propertyName string) (bool, error) {
+	prop, ok := properties[propertyName]
+	if !ok {
+		return false, fmt.Errorf("Property %s not found", propertyName)
+	}
+	propValue, ok := prop.(bool)
+	if !ok {
+		return false, fmt.Errorf("Property %s (%T) cannot be converted to bool", propertyName, prop)
+	}
+	return propValue, nil
+}
+
+func getServiceCheckStatus(activeState string) metrics.ServiceCheckStatus {
+	switch activeState {
+	case "active":
+		return metrics.ServiceCheckOK
+	case "inactive", "failed":
+		return metrics.ServiceCheckCritical
+	case "activating", "deactivating":
+		return metrics.ServiceCheckUnknown
+	}
+	return metrics.ServiceCheckUnknown
+}
+
+// isMonitored verifies if a unit should be monitored.
+func (c *SystemdCheck) isMonitored(unitName string) bool {
+	if len(c.config.instance.UnitNames) == 0 && len(c.config.instance.UnitRegexPatterns) == 0 {
+		return true
+	}
+	for _, name := range c.config.instance.UnitNames {
+		if name == unitName {
+			return true
+		}
+	}
+	for _, pattern := range c.config.instance.UnitRegexPatterns {
+		if pattern.MatchString(unitName) {
+			return true
+		}
+	}
+	return false
+}
+
+// Configure configures the systemd checks
+func (c *SystemdCheck) Configure(rawInstance integration.Data, rawInitConfig integration.Data) error {
+	err := c.CommonConfigure(rawInstance)
+	if err != nil {
+		return err
+	}
+	err = yaml.Unmarshal(rawInitConfig, &c.config.initConf)
+	if err != nil {
+		return err
+	}
+	err = yaml.Unmarshal(rawInstance, &c.config.instance)
+	if err != nil {
+		return err
+	}
+
+	for _, regexString := range c.config.instance.UnitRegexStrings {
+		pattern, err := regexp.Compile(regexString)
+		if err != nil {
+			c.Warnf("Failed to parse systemd check option unit_regexes: %s", err)
```

Comment:
note, just so you know: warnings set through `CheckBase.Warnf` are "cleared"  after each check run. Since we're in `Configure` here, which is only run once when the check is configured, this means that this warning will only be shown on the status page during the first check run of the check, and will then disappear.

Not a huge problem, but if you want to make sure the user sees this warning on the status page even after the 2nd+ check run, you'll want to set this warning at every check run.

(to be honest, I don't think this needs to be fixed now, but something to keep in mind)

Reply:
Seems for this case about configuration issues, just `log.Warnf` is more suitable, since the user would be probably looking at the logs when changing the config.

Reply:
Just pushed that change.